src/lex.cpp

```diff
@@ -662,7 +662,13 @@ void lexer::parse_number() {
         source_code_span(garbage_begin, garbage_end));
     input = garbage_end;
   };
-
+  input = this->parse_underscore(input);
+  bool has_underscore = *input == '_';
```

Comment:
When is `has_underscore` true? The code seems to work without lines 666-671. What is the purpose of these lines? I thought `parse_underscore` has its own loop, so why are we calling it again?

Reply:
Yeah, it's unnecessary.